Training/Training/file.c

```diff
@@ -1,13 +1,169 @@
+      if (num > 4 || num < 1)
+         printf (ANSI_COLOR_RED"Please enter a number between 1-4\n"ANSI_RESET_ALL);
+      else {
+         switch (num) {
+            case 1:
+               SearchAndSort ();
+               break;
+            case 2:
+               TestCases ();
+               break;
+            case 3:
+               ClrScreen ();
+               break;
+            case 4:
+               return 0;
+         }
+      }
+   }
+   return 0;
+}
+
+void BubbleSort (int arr[], int size) {
+   for (int i = 0; i < size - 1; i++) {
+      for (int j = 0; j < size - i - 1; j++) {
+         if (arr[j] > arr[j + 1]) {
+            arr[j] = arr[j] + arr[j + 1];
+            arr[j + 1] = arr[j] - arr[j + 1];
+            arr[j] = arr[j] - arr[j + 1];
+         }
+      }
+   }
+}
+
+int BinarySearch (int sortArr[], int key, int size) {
+   int low = 0, high = size - 1, result = -1;
+   while (low <= high) {
+      int mid = low + (high - low) / 2;
+      if (sortArr[mid] == key) {
+         result = mid;
+         high = mid - 1;
+      }
+      if (sortArr[mid] < key) low = mid + 1;
+      else high = mid - 1;
+   }
+   return result;
+}
+
+int GetUserInput (char* prompt) {
+   int64_t number = 0;
+   for (;;) {
+      char strNum[MAX_DIGIT];
+      char* endPtr = NULL;
+      printf ("%s", prompt);
+      char* strPtr = fgets (strNum, sizeof (strNum), stdin);
+      char* validInput = strchr (strNum, '\n');
+      number = strtoll (strNum, &endPtr, 10);
+      if (validInput == NULL)  while (getchar () != '\n');
+      if (strPtr == NULL || validInput == NULL || *endPtr != '\n' || (isdigit (strNum[0]) == 0 && strNum[0] != '-') || number < INT32_MIN || number > INT32_MAX) {
+         printf (ANSI_COLOR_RED"Invalid Number!!! Please re-enter\n"ANSI_RESET_ALL);
+         continue;
+      }
+      break;
+   }
+   return (int)number;
+}
+
+void SearchAndSort () {
+   int size = 0;
+   do {
+      size = GetUserInput ("\nEnter the size of array:");
+      if (size <= 0)  printf (ANSI_COLOR_RED"Invalid Number!!! Please re-enter\n"ANSI_RESET_ALL);
+   } while (size <= 0);
+   int* array = (int*)malloc (size * sizeof (int));
+   if (array == NULL) {
+      printf ("Memory allocation failed!\n");
+      return;
+   }
+   printf ("\nEnter the elements of the array:\n");
+   for (int i = 0; i < size; i++) {
+      printf ("[%d]", i);
+      array[i] = GetUserInput ("");
+   }
+   BubbleSort (array, size);
+   printf ("Sorted Array: ");
+   for (int i = 0; i < size; i++) printf (ANSI_COLOR_YELLOW"%d "ANSI_RESET_ALL, array[i]);
+   int index = BinarySearch (array, GetUserInput ("\n\nWhich element's index do you want to find?\n"), size);
+   printf ((index == -1) ? ANSI_COLOR_GREEN"Element is not present in array\n\n"ANSI_RESET_ALL : ANSI_COLOR_GREEN"Element is present at index %d\n\n"ANSI_RESET_ALL, index);
+   free (array);
+}
+
+void TestCases () {
```

Comment:
The presentation of the test case results is good. You can further develop it by representing it in a table format.